lib/go-atscfg/remapdotconfig_test.go

```diff
@@ -72,12 +91,203 @@ func pluginsFromTokens(tokens []string, prefix string) []string {
+					SetNumber: 1,
+					Pattern:   customRex,
+				},
+			},
+		}
+		dsRegexes = append(dsRegexes, dsr)
+	}
+	serverParams := []tc.Parameter{
+		tc.Parameter{
+			Name:       "trafficserver",
+			ConfigFile: "package",
+			Value:      "9",
+			Profiles:   []byte(`["global"]`),
+		},
+	}
+	remapConfigParams := []tc.Parameter{
+		tc.Parameter{
+			Name:       "cachekey.pparam",
+			ConfigFile: "remap.config",
+			Value:      "--cachekeykey=cachekeyval",
+			Profiles:   []byte(`["dsprofile"]`),
+		},
+		tc.Parameter{
+			Name:       "not_location",
+			ConfigFile: "cachekey.config",
+			Value:      "notinconfig",
+			Profiles:   []byte(`["global"]`),
+		},
+	}
+	cdn := &tc.CDN{
+		DomainName: "cdndomain.example",
+		Name:       "my-cdn-name",
+	}
+	topologies := []tc.Topology{}
+	cgs := []tc.CacheGroupNullable{}
+	serverCapabilities := map[int]map[ServerCapability]struct{}{}
+	dsRequiredCapabilities := map[int]map[ServerCapability]struct{}{}
+	configDir := `/opt/trafficserver/etc/trafficserver`
+
+	cfg, err := MakeRemapDotConfig(server, servers, dses, dss, dsRegexes, serverParams, cdn, remapConfigParams, topologies, cgs, serverCapabilities, dsRequiredCapabilities, configDir, &RemapDotConfigOpts{HdrComment: hdr})
+	if err != nil {
+		t.Fatal(err)
+	}
+	txt := cfg.Text
+
+	//t.Logf("text: %v", txt)
+
+	txt = strings.TrimSpace(txt)
+
+	testComment(t, txt, hdr)
+
+	txtLines := strings.Split(txt, "\n")
+	for _, line := range txtLines[2:] {
+		switch {
+		case strings.Contains(line, "http://dnsroutingname.mypattern1"):
+			mappings["http://dnsroutingname.mypattern1"] = true
+		case strings.Contains(line, "http://myregexpattern1"):
+			mappings["http://myregexpattern1"] = true
+		case strings.Contains(line, "http://server.mypattern0"):
+			mappings["http://server.mypattern0"] = true
+		case strings.Contains(line, "https://server.mypattern0"):
+			mappings["https://server.mypattern0"] = true
+		case strings.Contains(line, "http://mcastserver1.mypattern0"):
+			mappings["http://mcastserver1.mypattern0"] = true
+		case strings.Contains(line, "https://mcastserver1.mypattern0"):
+			mappings["https://mcastserver1.mypattern0"] = true
+		case strings.Contains(line, "http://mcastserver2.mypattern0"):
+			mappings["http://mcastserver2.mypattern0"] = true
+		case strings.Contains(line, "https://mcastserver2.mypattern0"):
+			mappings["https://mcastserver2.mypattern0"] = true
+		case strings.Contains(line, "http://myregexpattern0"):
+			mappings["http://myregexpattern0"] = true
+		case strings.Contains(line, "https://myregexpattern0"):
+			mappings["https://myregexpattern0"] = true
+		default:
+			t.Fatalf("unexpected remap line '%v'", line)
+		}
+	}
+	for key, val := range mappings {
+		if !val {
+			t.Fatalf("expected to find remap rule for '%v'", key)
+		}
+	}
+}
 func TestMakeRemapDotConfig0(t *testing.T) {
 	hdr := "myHeaderComment"
 
 	server := makeTestRemapServer()
 	server.Type = "EDGE"
 
+	servers := makeTestAnyCastServers()
```

Comment:
would it be better to make some/most of the normal tests use an empty collection for this?